diff --git a/linux-user/syscall.c b/linux-user/syscall.c
index 0f5b139..f764a17 100644
--- a/linux-user/syscall.c
+++ b/linux-user/syscall.c
@@ -2038,8 +2038,6 @@ static abi_long do_socket(int domain, int type, int protocol)
         return ret;
     }

-    if (domain == PF_NETLINK)
-        return -TARGET_EAFNOSUPPORT;
     ret = get_errno(socket(domain, type, protocol));
     if (ret >= 0) {
         ret = sock_flags_fixup(ret, target_type);
